From 9f5da322efc54f56eb0531888b65293ffee46fa1 Mon Sep 17 00:00:00 2001
From: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Date: Mon, 18 Jul 2016 14:03:03 +0530
Subject: [PATCH 0728/1566] uio: apm: Fix warnings

Fix the below warnings

drivers/uio/uio_xilinx_apm.c: In function 'xapm_probe':
drivers/uio/uio_xilinx_apm.c:283:24: warning: passing argument 1 of 'clk_disable_unprepare' from incompatible pointer type
  clk_disable_unprepare(&xapm->param.clk);
                        ^
In file included from drivers/uio/uio_xilinx_apm.c:26:0:
include/linux/clk.h:482:20: note: expected 'struct clk *' but argument is of type 'struct clk **'
 static inline void clk_disable_unprepare(struct clk *clk)
                    ^
drivers/uio/uio_xilinx_apm.c: In function 'xapm_remove':
drivers/uio/uio_xilinx_apm.c:298:24: warning: passing argument 1 of 'clk_disable_unprepare' from incompatible pointer type
  clk_disable_unprepare(&xapm->param.clk);
                        ^
In file included from drivers/uio/uio_xilinx_apm.c:26:0:
include/linux/clk.h:482:20: note: expected 'struct clk *' but argument is of type 'struct clk **'
 static inline void clk_disable_unprepare(struct clk *clk)

Signed-off-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/uio/uio_xilinx_apm.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/uio/uio_xilinx_apm.c b/drivers/uio/uio_xilinx_apm.c
index 5d5d1fe..194dd88 100644
--- a/drivers/uio/uio_xilinx_apm.c
+++ b/drivers/uio/uio_xilinx_apm.c
@@ -277,7 +277,7 @@ static int xapm_probe(struct platform_device *pdev)
 	return 0;
 
 err_clk_dis:
-	clk_disable_unprepare(&xapm->param.clk);
+	clk_disable_unprepare(xapm->param.clk);
 	return ret;
 }
 
@@ -292,7 +292,7 @@ static int xapm_remove(struct platform_device *pdev)
 	struct xapm_dev *xapm = platform_get_drvdata(pdev);
 
 	uio_unregister_device(&xapm->info);
-	clk_disable_unprepare(&xapm->param.clk);
+	clk_disable_unprepare(xapm->param.clk);
 
 	return 0;
 }
-- 
1.7.5.4

